### DISCUSSION TITLE
Excluding generated code from coverage (moved here from issues)

### DISCUSSION THREAD

**Discussion by Zegveld** [mapstruct/mapstruct#2769] [General]
Original post by @Payne911 at issue #1574
> My team would love such a feature as well. We are trying to exclude the generated classes from the unit-test code coverage stats in our pipeline via a custom annotation. It would be great to be able to propage that annotation to the implementation of the mappers.
> 
> But maybe another way to approach this would be for MapStruct to have its own `@Generated` class and to change the Retention Policy from `SOURCE` to either `RUNTIME` or `CLASS`, as explained [there](https://www.baeldung.com/jacoco-report-exclude#excluding-with-custom-annotation). That is how Lombok does it.

@Payne911, why not have the generated classes live in a second source-folder which is not analyzed for coverage?

**Comment by payne911** [comment]
Excluding the folder is our current approach, but the infra team has specific requirements which prevents us from using Jacoco exclusions, and thus requires us to exclude the generated classes with an annotation.

**Comment by akshatprakash** [comment]
lombok has a way by annotating classes with custom annotation @Generated using this flag lombok.addLombokGeneratedAnnotation=true

i just checked your PR's and saw @AnnotateWith( MyCustomAnnotation.class ) is coming up in 1.6 how can i use it i don't see it in maven repo

**Reply by filiphr** [reply]
@akshatprakash you would be able to use it by defining your own annotation and adding `@AnnotateWith(MyCustomAnnotation.class)` to the mapper.

Nothing is stopping you in adding a custom `@Generated` annotation in your own code.
